peer/roundrobin/peerring.go

```diff
@@ -41,22 +43,35 @@ type peerRing struct {
 	nextNode   *ring.Ring
 }
 
-// GetPeer returns the Peer from the Ring or Nil
-func (pr *peerRing) GetPeer(pid peer.Identifier) peer.Peer {
-	node, ok := pr.peerToNode[pid.Identifier()]
-	if !ok {
+type ringSubscriber struct{}
+
+func (r ringSubscriber) NotifyStatusChanged(peer.Identifier) {}
+
+var nopSubscriber = ringSubscriber{}
```

Comment:
Could use the `node` as the subscriber instead. With that, `Remove(pid, sub)` would be able to to skip looking up the node.